source/blog/2016-10-04-ovirt-provider-ovn.md

```diff
@@ -0,0 +1,125 @@
+
+    `yum install ovirt-provider-ovn-0-1.noarch.rpm`
+
+OVN requires OVN to be installed on the host (version 2.6 or later). The following OVN packages must be installed:
+
+* `openvswitch`
+* `openvswitch-ovn-common`
+* `openvswitch-ovn-central`
+* `python-openvswitch`
+
+These are also available from the [OVS website](http://openvswitch.org/download/) or built using the code downloaded from the OVS repo (described below).
+
+After installation, the provider can be started as follows:
+
+	`systemctl start ovirt-provider-ovn`
+
+The provider can then be added to oVirt as an external network provider. In order to add a new provider, go to the External Providers section in the oVirt UI and click the Add button.
+
+![adding a new provider](uarwo40-edit-cluster.png)
+
+When the provider is successfully added, existing OVN networks can be imported to oVirt.
+New OVN networks can be defined using oVirt by adding a network and specifying it to be added on an external provider (make sure you add the external provider in read-write mode, so that you can add external networks from oVirt).
+
+A VM NIC can be added to OVN networks by simply choosing an external network during NIC provisioning.
+
+OVN based networking brings many advantages to oVirt:
+* More granular security - it brings complete network isolation without the need for defining VLANs.
+* Less resources - since the OVN networks are logical overlays, many of them can be defined on the same underlying physical infrastructure. As oposed to traditional oVirt networking, where each network required a separate host NIC, all the OVN networks can be acommodated on a single NIC.
+* Easier management - the management of the networking infrastructure should also become much easier, as instead of managing the network isolation on each of the networking components, it will be automatically taken care of by OVN.
+
+## Building oVirt OVN Provider rpms From Repository
+
+Clone the repository:
+
+`git clone https://github.com/openvswitch/ovirt-provider-ovn`
+
+Build the rpms:
+
+`cd ovirt-provider-ovn`
+`make rpm`
+
+The built rpms can be found in: `~/rpmbuild/RPMS/noarch/`
+
+## Building OVN Provider rpms From Repository
+
+Clone the repository:
+
+`git clone https://github.com/openvswitch/ovs`
+
+Install the following packages, as they are need to build ovn:
+
+`dnf -y install gcc make python-devel openssl-devel kernel-devel graphviz kernel-debug-devel autoconf automake rpm-build redhat-rpm-config rpm-build rpmdevtools bash-completion autoconf automake libtool PyQt4 groff libcap-ng-devel python-twisted-core python-zope-interface graphviz openssl-devel selinux-policy-devel`
+
+Build the ovn rpms:
+
+`cd ovs`
+`./boot.sh`
+`./configure`
+`make dist`
+`cp openvswitch-<version>.tar.gz $HOME/rpmbuild/SOURCES`
+`cd $HOME/rpmbuild/SOURCES`
+`tar xzf openvswitch-<version>.tar.gz`
+`cd openvswitch-<version>`
+`rpmbuild -bb rhel/openvswitch-fedora.spec` 
+
+The built rpms will reside here: `~/rpmbuild/RPMS/x86_64/`
+
```

Comment:
Added:

The command above should create OVN tunnels to other OVN controllers (if at least one other ovn-controller is present).
Please verify that the tunnel has been created by issueing the following commands:

`ip link` - the result should include a link called `genev_sys_ ...`

`ovs-vsctl show` - the bridge `br-int` should include a port looking somewhat like this:

    Port "ovn-6b2d2f-0"
       Interface "ovn-6b2d2f-0"
            type: geneve
            options: {csum="true", key=flow, remote_ip="10.35.128.14"}


Please check the OVN logs in case of problems. The relevant OVN logs are located in `/var/log/openvswitch/`.